File: SuperMarket.Web/SuperMarket.Web/wwwroot/Scripts/Models/Product.js
class Product extends Entity
{
    constructor(json)
    {
        super( json);

        if (json)
        {
            this.ProductName = json.productName;
            this.Price = json.price;
        }
        else
        {
            this.ProductName = "";
            this.Price = "";
        }
    }
}

File: SuperMarket.Lib/Models/Product.cs
using SuperMarket.Lib.Core;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace SuperMarket.Lib.Models
{
    public class Product : Entity
    {
        public string ProductName { get; set; }
        public int Price { get; set; }

        [JsonIgnore]
        public ICollection<OrderItem> OrderItems { get; set; }
    }
}
